frontend/cypress/integration/profile.spec.js

```diff
@@ -50,4 +55,235 @@ describe('Profile', () => {
+        .click();
+
+      cy.get('[data-test-id="suspend-button"]')
+        .should('not.be.disabled');
+
+      cy.get('[data-test-id="radio-duration-week"]')
+        .click();
+
+      cy.get('[data-test-id="radio-duration-other-input"]')
+        .should('not.exist');
+    });
+
+    const checkSuspension = (reason, duration, getStartDate, getEndDate) => {
+      const start = getStartDate();
+      cy.get('[data-test-id="suspend-redirection-button"]')
+        .click();
+
+      cy.get(`[data-test-id="radio-reason-${reason}"]`)
+        .click();
+
+      cy.get(`[data-test-id="radio-duration-${duration}"]`)
+        .click();
+
+      cy.get('[data-test-id="suspend-button"]')
+        .click();
+
+      cy.wait('@suspend').then(response => {
+        const end = getEndDate();
+
+        cy.wrap(response.request.body.reason).should('eq', reason);
+        cy.wrap((new Date(response.request.body.date)).getTime() < end.getTime()).should('eq', true);
+        cy.wrap((new Date(response.request.body.date)).getTime() > start.getTime()).should('eq', true);
+      });
+    };
+
+    it('should suspend profile for a week because of too much request', () => {
+      const addWeek = () => {
+        const date = new Date();
+        date.setDate(date.getDate() + 7);
+        return date;
+      };
+      checkSuspension('toomuch', 'week', addWeek, addWeek);
+    });
+
+    it('should suspend profile for a month because of holidays', () => {
+      const addMonth = () => {
+        const date = new Date();
+        date.setMonth(date.getMonth() + 1);
+        return date;
+      };
+      checkSuspension('holidays', 'month', addMonth, addMonth);
+    });
+
+    it('should suspend profile forever because of reimbursments not there', () => {
+      checkSuspension('reimbursments', 'forever', () => new Date('9999/12/30'), () => new Date('10000/01/01'));
+    });
+
+    it('should suspend profile forever because of convention not signed', () => {
+      checkSuspension('convention', 'forever', () => new Date('9999/12/30'), () => new Date('10000/01/01'));
+    });
+
+    it('should suspend profil for custom reasons and custom date', () => {
+      cy.get('[data-test-id="suspend-redirection-button"]')
+        .click();
+
+      cy.get('[data-test-id="radio-reason-other"]')
+        .click();
+      cy.get('[data-test-id="radio-reason-other-input"]')
+        .type('parcequuuuuuuuuue');
+
+      cy.get('[data-test-id="radio-duration-other"]')
+        .click();
+
+      cy.get('[data-test-id="radio-duration-other-input"]')
+        .click();
+
+      const tomorrow = new Date();
+      tomorrow.setDate(tomorrow.getDate() + 2);
+
+      cy.get(`.react-datepicker__day--0${tomorrow.getDate()}`)
+        .click();
+
+      cy.get('[data-test-id="suspend-button"]')
+        .click();
+
+      cy.wait('@suspend').then(response => {
+        cy.wrap(response.request.body.reason).should('eq', 'other: parcequuuuuuuuuue');
```

Comment:
Does it mean that if we print somewhere, let's say a metabase question, it would be print like this? Shouldn't we put 'Autre: parcequuuuuuuuuue'?